tests/test_connection.py

```diff
@@ -271,6 +272,30 @@ def mock_request(*args, **kwargs):
         with pytest.raises(RequestException, match=r"Failed to resolve hostname 'test.xanadu.ai'"):
             connection.request("GET", "/healthz")
 
+    def test_request_headers(self, connection):
+        """Test that request() passes the headers attribute to _request if the headers argument
+        is not provided."""
+
+        connection._request = MagicMock()
+        connection.request(method="method", path="path")
+
+        connection._request.assert_called_once_with(
+            method="method", url=connection.url("path"), headers=connection.headers
+        )
+
+    @pytest.mark.parametrize("extra_headers", [{"X-Test": "data"}, {}])
+    def test_request_extra_headers(self, connection, extra_headers):
+        """Tests that request() passes the combined headers from the headers attribute
+        and the headers argument to the _request method."""
+        connection._request = MagicMock()
+
+        connection.request(method="post", path="path", headers=extra_headers)
+        connection._request.assert_called_once_with(
+            method="post",
+            url=connection.url("path"),
+            headers={**connection.headers, **extra_headers},
+        )
```

Comment:
I like the coverage of these tests but we should consider using [`responses`](https://pypi.org/project/responses/) to avoid coupling the implementation of `xcc.Connection.request()` to the usage of a private helper function.

Reply:
the tests now use responses